### PR TITLE
New Widget: Knobs

### PR DESCRIPTION
**Pull request by PROP65**
Adds `int`/`float` knob widgets
knob
It can be dragged with the mouse, moved with scroll wheel, or moved with the up/down arrows.
```c
/* API */
NK_API nk_bool nk_knob_float(struct nk_context*, float min, float *val, float max, float step, enum nk_heading zero_direction, float dead_zone_degrees);
NK_API nk_bool nk_knob_int(struct nk_context*, int min, int *val, int max, int step, enum nk_heading zero_direction, float dead_zone_degrees);
```
```c
/* Usage */
nk_labelf(ctx, NK_TEXT_LEFT, "Knob float: %.2f", float_knob);
nk_knob_float(ctx, 0, &float_knob, 5.0, 0.5f, NK_DOWN, 60.0f);
```

### CODE REVIEW

**Comment by RobLoach** [comment]
This is fantastic. Nicely done.

File: src/nuklear_knob.c [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        knob_value = NK_CLAMP(knob_min, knob_value, knob_max);
+    }
+
+    /* knob widget state */
+    if (nk_input_is_mouse_hovering_rect(in, bounds)){
+        *state = NK_WIDGET_STATE_HOVERED;
+        if (in) {
+            /* handle scroll and arrow inputs */
+            if (in->mouse.scroll_delta.y > 0 ||
+               (in->keyboard.keys[NK_KEY_UP].down && in->keyboard.keys[NK_KEY_UP].clicked))
```
Comment (RobLoach) [inline src/nuklear_knob.c:57]: Mouse hovering with key state changes isn't a user interaction behavior across any of the other widgets. I'm conflicted on this one.

**Comment by RobLoach** [comment]
@PROP65 I added the different styles so that it's still usable in the different default color schemes. Is there anything else you'd like to add before we merge this in?

Screenshot from 2024-08-27 11-45-01

**Comment by PROP65** [comment]
LGTM

**Pull request merged**